File: Soultion/167-Two-Sum-II---Input-Array-Is-Sorted.dart
class Solution {
  List<int> twoSum(List<int> numbers, int target) {
    int left = 0;
    int right = numbers.length - 1;

    while (left < right) {
      int sum = numbers[right] + numbers[left];
      if (sum == target) {
        return [left + 1, right + 1];
      }
      if (sum > target) {
        right--;
      } else {
        left++;
      }
    }
    return [0];
  }
}

File: Soultion/167-Two-Sum-II-Input-Array-Is-Sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int r = numbers.size() - 1;
        int l = 0;
        while (l < r) {
            if (numbers[r] + numbers[l] > target) {
                r--;
            }
            else if (numbers[r] + numbers[l] < target) { 
                l++;
            }
            else {
                return {l + 1, r + 1};  
            }
        }
        return {};
    }
};
